lib/status.go

```diff
@@ -97,19 +97,45 @@ func (s *StatusSender) setLastUnsentStatus(status *WorkerStatus) {
 	s.lastUnsentStatus = status
 }
 
-// SafeSendStatus persists status before sending it
+// SafeSendStatus persists status before sending it, it will try to send status
+// until successfully.
 func (s *StatusSender) SafeSendStatus(ctx context.Context, status WorkerStatus) error {
-	if err := s.workerMetaClient.Store(ctx, s.workerID, &status); err != nil {
-		return err
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.Trace(ctx.Err())
+		default:
+		}
+		err := s.SyncSendStatus(ctx, status)
```

Comment:
max retry time should be set here?

Reply:
added default timeout

Reply:
where is timeout ?

Reply:
It is mixed in context
https://github.com/hanfei1991/microcosm/blob/af60dd6d7fd4b43cd2137219da168364e8d6c7d2/lib/status.go#L107-L116